src/common/reconciler.go

```diff
@@ -229,58 +202,83 @@ func (r *ServiceReconciler) updateService(service *opslevel.Service, registratio
 		log.Warn().Msgf("[%s] unexpected happened: service passed to be updated is nil", registration.Name)
 		return
 	}
-	updateServiceInput := opslevel.ServiceUpdateInput{
-		Id:          &service.Id,
-		Product:     opslevel.RefOf[string](registration.Product),
-		Description: opslevel.RefOf[string](registration.Description),
-		Language:    opslevel.RefOf[string](registration.Language),
-		Framework:   opslevel.RefOf[string](registration.Framework),
+
+	// updateServiceInput contains the changes needed to reconcile the service
+	updateServiceInput := opslevel.ServiceUpdateInput{Id: &service.Id}
+	// for each field - check if the value exists in the registration AND if the value has changed compared to what is currently set
+	if registration.Description != "" && registration.Description != service.Description {
```

Comment:
won't the `cmp.Diff` catch the 2nd part of these logic checks?  IE `registration.Description != service.Description` seems unnecessary given the `cmp.Diff` lower down.

Reply:
@rocktavious `cmp.Diff()` returns a string that is only useful to:

1. Check if there is a diff at all
2. Show the user the diff in an easy to read format similar got `git diff`

I don't see how we can use it on single fields, can you elaborate?

Reply:
@rocktavious 

```
if cmp.Equal(opslevel.ServiceUpdateInput{Id: &service.Id}, updateServiceInput) {
		log.Info().Msgf("[%s] No changes detected to fields - skipping update", service.Name)
		return
	}
```

I have changed the line that determines whether or not to continue based on if there is a diff or not to use `cmp.Equal` instead of `cmp.Diff` since it makes more sense here like you point out. 

Also added a comment at the start of the func explaining why we need to individually compare fields.